### PR TITLE
write cctrl feature

### PR DESCRIPTION
**Pull request by aleksandarbos**
## Description (e.g. "Related to ...", "Closes ...", etc.)

- added support for adding content controls
- enabled access towards content controls within header and footer parts

## Code review checklist

- [ ] Private platform tests at `core/tests/test_python-docx` are updated and passing
- [ ] If this change is going to be deployed on Cloudsmith after merge, python-docx version number should be increased 

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: docx/oxml/table.py [hunk 1]
```diff
@@ -759,6 +759,7 @@ class CT_TcPr(BaseOxmlElement):
         'w:cellDel', 'w:cellMerge', 'w:tcPrChange'
     )
     tcW = ZeroOrOne('w:tcW', successors=_tag_seq[2:])
+    tcBorders = ZeroOrOne('w:tcBorders', successors=_tag_seq[6:])
```
Comment (balsa-sarenac) [inline docx/oxml/table.py:762]: Why do we need to extend this?

Comment (balsa-sarenac) [inline docx/oxml/table.py:762]: Why do we need this?

File: docx/text/paragraph.py [hunk 2]
```diff
@@ -39,6 +41,77 @@ def add_run(self, text=None, style=None):
...
 
+    def add_field(self, instrText=None):
+        """
+        Adds new field `w:fldChar` to run. Pass `instrText` to specify
+        filed instruction.
+        """
+        self.add_run().add_fldChar()
+        if instrText:
+            self.add_run().add_instrText(instrText)
+        self.add_run().add_fldChar(fldCharType='end')
```
Comment (balsa-sarenac) [inline docx/text/paragraph.py:52]: Maybe this should not be part of this pull request

File: docx/text/paragraph.py [hunk 3]
```diff
@@ -39,6 +41,77 @@ def add_run(self, text=None, style=None):
...
+        alias = sdtPr._add_alias()
+        alias.set('{%s}val' % nsmap['w'], alias_name)
+
+        temp = sdtPr._add_temporary()
+        temp.set('{%s}val' % nsmap['w'], temporary)
+
+        sdtContent = sdt._add_sdtContent()
+
+        if not text:
+            r = sdtContent._add_r()
```
Comment (balsa-sarenac) [inline docx/text/paragraph.py:92]: Would it make sense to move this line before if?

File: docx/text/paragraph.py [hunk 4]
```diff
@@ -39,6 +41,77 @@ def add_run(self, text=None, style=None):
...
+            rPr = r._add_rPr()
+            rStyle = rPr._add_rStyle()
+            rStyle.set('{%s}val' % nsmap['w'], 'PlaceholderText')
+            rPr._add_b()
+            rPr._add_bCs()
+            t = r._add_t()
+            placeholder_txt = placeholder_txt or 'Click or tap here to enter text'
+            t.text = placeholder_txt
+            active_placeholder = sdtPr._add_active_placeholder()
+            active_placeholder.set('{%s}val' % nsmap['w'], 'true')
```
Comment (balsa-sarenac) [inline docx/text/paragraph.py:102]: Would it make sense to move this to a separate function?

**Review by balsa-sarenac — COMMENTED** [review]
Looks good, left a couple of comments.

File: docx/oxml/table.py (hunk 1 above)
Comment (aleksandarbos) [reply docx/oxml/table.py:762]: we don't it was probably a leftover from different branch, thanks!

File: docx/text/paragraph.py (hunk 2 above)
Comment (aleksandarbos) [reply docx/text/paragraph.py:52]: yes, same as previous, good catch

File: docx/text/paragraph.py (hunk 3 above)
Comment (aleksandarbos) [reply docx/text/paragraph.py:92]: yes it repeats itself

File: docx/text/paragraph.py (hunk 4 above)
Comment (aleksandarbos) [reply docx/text/paragraph.py:102]: sure

**Review by balsa-sarenac — APPROVED** [review]
Looks good to me!
